app/particle/particle_simulation.py

```diff
@@ -63,87 +63,112 @@
         help = "force constant")
 args = parser.parse_args()
 
-params.num_particles = args.numparticles if args.numparticles else 100
 params.simulation_height = args.height if args.height else 1000
 params.simulation_width = args.width if args.width else 1000
 params.simulation_depth = args.depth if args.depth else 1000
-params.dt = args.dt if args.dt else 0.0005
-#params.force = args.force if args.force else 100
-params.force = args.force if args.force else 100000
+
+params.num_particles = args.numparticles if args.numparticles else 100
+params.new_num_particles = params.num_particles
+params.max_radius = min(params.simulation_width, params.simulation_height, params.simulation_depth)//32
+
 params.num_active_workers = 0
 params.new_num_active_workers = 0
 params.partitions = {}
-params.max_radius = min(params.simulation_width, params.simulation_height, params.simulation_depth)//32
+
+params.dt = args.dt if args.dt else 0.0005
 params.timesteps_per_second = 0
+params.force = args.force if args.force else 100000
+
 params.init_total_energy = 0.0
 params.curr_total_energy = 0.0
 
+def add_one_particle(particle_id):
+    radius = 30
+    position = [random.randint(radius, params.simulation_width - 1),
+                random.randint(radius, params.simulation_height - 1),
+                random.randint(radius, params.simulation_depth - 1)]
+    velocity = [400*random.randint(0,radius//4),
+                400*random.randint(0,radius//4),
+                400*random.randint(0,radius//4)]
+    mass = 3
+    params.init_total_energy += 0.5 * mass * (velocity[0]**2 + velocity[1]**2 + velocity[2]**2)
+    thread_num = util.determine_particle_thread_num(position[0])
+    new_particle = Particle(particle_id, thread_num, position, velocity, mass, radius)
+    params.partitions[thread_num].add_particle(new_particle)
+
+def subtract_one_particle(particle_id):
+    if params.rank is 0:
+        for i in range(1, params.num_active_workers+1):
```

Comment:
Also, you're not using the variable `i` here in the body of the loop.  Convention indicates that this would be better written as
```
for _ in range(...)
```